gui/guiComponent.py

```diff
@@ -0,0 +1,158 @@
+import os
+from tkinter import *
+import screeninfo
+import time
+from utils.config import USE_GUI, TIMEOUT_GUI
+
+# We do not need this on Linux.
+if os.name == "nt":
+    import pythoncom
+    import win32com.client
+    import win32gui
+
+
+def windowEnumerationHandler(hwnd, top_windows):
+    """
+    Handler for Windows OS enumeration of all open windows.
+    Used to return to the Path of Exile window after displaying the overlay.
+    """
+    top_windows.append((hwnd, win32gui.GetWindowText(hwnd)))
+
+
+def windowToFront(root):
+    # This is necessary for displaying the GUI window above active window(s) on the Windows OS
+    if os.name == "nt":
+        # In order to prevent SetForegroundWindow from erroring, we must satisfy the requirements
+        # listed here:
+        # https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setforegroundwindow
+        # We satisfy this by internally sending the alt character so that Windows believes we are
+        # an active window.
+        # We need this pythoncom call for win32com use in a thread.
+        pythoncom.CoInitialize()
+        shell = win32com.client.Dispatch("WScript.Shell")
+        shell.SendKeys("%")
+        win32gui.SetForegroundWindow(root.winfo_id())
+
+
+def windowRefocus(name):
+    """
+    Restore focus to a window, if on Windows.
+    TODO: If originating window was NOT "name", return to previous window.
+    """
+
+    if os.name == "nt":
+        results = []
+        top_windows = []
+        win32gui.EnumWindows(windowEnumerationHandler, top_windows)
+        for i in top_windows:
+            if name == i[1].lower():
+                win32gui.ShowWindow(i[0], 5)
+                win32gui.SetForegroundWindow(i[0])
+                break
+
+components = []
+
+class GuiComponent:
+    def __init__(self):
+        self.frame = None
+        self.closed = True
+        self.opened = time.time()
+        self.elapsed = 0
+        self.have_timeout = False
+        components.append(self)
+    def should_close(self):
+        if self.is_closed():
+            return
+        if not self.have_timeout:
+            return
+        self.elapsed = time.time() - self.opened
+        if self.elapsed >= int(TIMEOUT_GUI):
+            elapsed = 0
+            self.close()
+    
+    def prepare_window(self):
+        tk = Tk().withdraw()
+        frame = Toplevel()
+        frame.overrideredirect(True)
+        frame.option_add("*Font", "courier 12")
+        frame.withdraw()
+        self.frame = frame
+
+    def is_closed(self):
+        if self.closed:
+            return True
+        return False
+
+    def close(self):
+        if self.closed:
+            return
+        self.closed = True
+        self.frame.destroy()
+        self.frame.update()
+        self.frame = None
+        windowRefocus("path of exile")
```

Comment:
This should not happen when closing previous window, only when its closed after delay